pkg/config/validation/validation.go

```diff
@@ -2581,6 +2595,14 @@ var ValidateServiceEntry = registerValidateFunc("ValidateServiceEntry",
 			return fmt.Errorf("cannot cast to service entry")
 		}
 
+		if err := validateAlphaWorkloadSelector(serviceEntry.WorkloadSelector); err != nil {
+			return err
+		}
+
+		if serviceEntry.WorkloadSelector != nil && serviceEntry.Endpoints != nil {
+			errs = appendErrors(errs, fmt.Errorf("service entry may not have selector and endpoints"))
```

Comment:
nit:  Please change the error message to `Only one of WorkloadSelector or Endpoints is allowed in Service Entry` to be more clearer